index.js

```diff
@@ -9,30 +9,22 @@ module.exports = {
   included: function() {
     this._super.included.apply(this, arguments);
 
-    // Default options
-    var defaultOptions = {
-      enabled: this.app.env === 'production'
-    };
-
-    // If the imagemin options key is set to `false` instead of
-    // an options object, disable the addon
-    if (typeof this.app.options.imagemin === 'boolean') {
-      this.options = this.app.options.imagemin = { enabled: this.app.options.imagemin };
-    } else {
-      this.options = this.app.options.imagemin = this.app.options.imagemin || {};
+    this.enabled = this.app.env === 'production'
+    this.options = this.app.options.imagemin = this.app.options.imagemin || {};
+
+    if ('enabled' in this.options) {
+      this.enabled = this.options.enabled;
+      delete this.options.enabled;
```

Comment:
That would be redundant. Validation is already handled by `broccoli-imagemin`, which will throw an error if you pass any value in the `enabled` property.